modules/sdk-coin-icp/src/icp.ts

```diff
@@ -98,7 +107,12 @@ export class Icp extends BaseCoin {
     if (!txHex) {
       throw new Error('txHex is required');
     }
-    const explainedTx = await this.explainTransaction({ txHex });
+
+    let signableHex = txPrebuild?.txInfo as string | undefined;
+    if (txPrebuild.txInfo && txPrebuild.txInfo !== undefined && typeof txPrebuild.txInfo === 'string') {
+      signableHex = txPrebuild?.txInfo as string;
+    }
+    const explainedTx = await this.explainTransaction({ txHex, signableHex });
```

Comment:
Here, you are either sending string or undefined to explainTransaction for signableHex.
As this is optional in explainTransaction(), you should only send it when it is valid string.

Reply:
You should something like this -
```suggestion
const params: { txHex: string; signableHex?: string } = { txHex };

    if (txPrebuild.txInfo && typeof txPrebuild.txInfo === 'string') {
  params.signableHex = txPrebuild.txInfo;
}

const explainedTx = await this.explainTransaction(params);
```

Reply:
used and interface